### PR TITLE
Feature/api boilerplate

### PR DESCRIPTION
**Pull request by toboid**
Basic API boilerplate. Let me know what you think about the version endpoint. It's a pattern from Eran Hammer. I think it's quite useful to be able to see the version and also to have an endpoint that we can use to check if the api is up, without any dependencies. I considered putting it in the swagger but not sure it really belongs there. Interested to know what you think.
Logging still needs looking into.

### CODE REVIEW

File: Gruntfile.js [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+          reporter: 'spec',
+          clearRequireCache: false
+        },
+        src: ['test/*.js']
+      }
+    },
+    watch: {
+      js: {
+        files: ['<%= jsFiles %>'],
+        tasks: ['lint', 'test']
```
Comment (brainsiq) [inline Gruntfile.js]: I think I prefer to run test first so linting errors don't stop broken tests from showing up.

**Comment by brainsiq** [comment]
I agree that version doesn't need to go into swagger as there would be separate swagger documentation of each version anyway.

Think you need to rebase to pick up the latest swagger.yaml changes I made. The JSON file I gave you already contains these changes however.

**Comment by brainsiq** [comment]
Should we run something like retire.js in there as well?

**Comment by toboid** [comment]
yeah retire.js makes sense. adding

File: Gruntfile.js (hunk 1 above)
Comment (toboid) [inline Gruntfile.js]: have updated this. also realised that linting was failing due to xo being ES5 only by default. This is now fixed.

File: .eslintrc [hunk 2]
```diff
@@ -0,0 +1,30 @@
+{
+  "extends": "xo-space",
+  "rules": {
+    "strict": [2, "global"],
+    "no-var": 1
```
Comment (toboid) [inline .eslintrc]: Have set this to warning (not error). There is a nice description [here](http://eslint.org/docs/rules/no-var). Any problems with this?

**Comment by toboid** [comment]
retire.js added

**Comment by brainsiq** [comment]
All good!

**Pull request merged**